fdbserver/BackupWorker.actor.cpp

```diff
@@ -1085,18 +1083,24 @@ ACTOR Future<Void> pullAsyncData(BackupData* self) {
 
 		// Note we aggressively peek (uncommitted) messages, but only committed
 		// messages/mutations will be flushed to disk/blob in uploadData().
-		int64_t peekedBytes = 0;
+		state int64_t peekedBytes = 0;
+		// Hold messages until we know how many we can take, self->messages always
+		// contains messages that have been reserved memory for. I.e., lock->release()
+		// will not encounter message that has not been reserved memory.
```

Comment:
```suggestion
		// Hold messages until we know how many we can take, self->messages always
		// contains messages that we have reserved memory for. Therefore, lock->release()
		// will always encounter message with reserved memory.
```

A bit more clear for me. Feel free to reword further.